spec/classes/broker_spec.rb

```diff
@@ -31,6 +36,10 @@
         context 'defaults' do
           it { is_expected.to contain_file('/opt/kafka/config/server.properties') }
         end
+        context 'with  manage_log4j => true' do
+          let(:params) { {'manage_log4j' => true} }
+          it { is_expected.to contain_file('/opt/kafka/config/log4j.properties') }
```

Comment:
You can test [resource parameters](https://rspec-puppet.com/documentation/classes/#test-resource-parameters) like so:

```suggestion
          it { is_expected.to contain_file('/opt/kafka/config/log4j.properties').with_content(%r{^log4j.appender.kafkaAppender.MaxFileSize=50MB$}) }
```

What you want to test is the content parameter of the file, and it's a regular String (the return value from the `epp()` call after applying the template).

You can pass non-default values in a context and make sure they are processed as expected.